src/js/control-bar/fullscreen-toggle.js

```diff
@@ -21,7 +25,20 @@ class FullscreenToggle extends Button {
   buildCSSClass() {
     return `vjs-fullscreen-control ${super.buildCSSClass()}`;
   }
-
+  /**
+   * Handles Fullscreenchange on the component and change control text accordingly
+   *
+   * @method handleFullscreenChange
+   */
+  handleFullscreenChange() {
+    let fst = this.player().controlBar.fullscreenToggle;
```

Comment:
This seems unreliable - making the change that @gkatsev suggests would mean that `fst` would be entirely replaced in this method by `this`, which would make  FullscreenToggle much more robust and extensible.